app/src/main/java/io/homeassistant/companion/android/settings/shortcuts/ManageShortcutsViewModel.kt

```diff
@@ -0,0 +1,314 @@
+package io.homeassistant.companion.android.settings.shortcuts
+
+import android.app.Application
+import android.content.Intent
+import android.content.pm.ShortcutInfo
+import android.content.pm.ShortcutManager
+import android.graphics.Bitmap
+import android.graphics.drawable.Drawable
+import android.graphics.drawable.Icon
+import android.os.Build
+import android.util.Log
+import android.widget.Toast
+import androidx.annotation.RequiresApi
+import androidx.appcompat.content.res.AppCompatResources
+import androidx.compose.runtime.mutableStateMapOf
+import androidx.compose.runtime.mutableStateOf
+import androidx.core.content.getSystemService
+import androidx.core.graphics.drawable.DrawableCompat
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.viewModelScope
+import com.maltaisn.icondialog.pack.IconPack
+import com.maltaisn.icondialog.pack.IconPackLoader
+import com.maltaisn.iconpack.mdi.createMaterialDesignIconPack
+import dagger.hilt.android.lifecycle.HiltViewModel
+import io.homeassistant.companion.android.common.R
+import io.homeassistant.companion.android.common.data.integration.Entity
+import io.homeassistant.companion.android.common.data.integration.IntegrationRepository
+import io.homeassistant.companion.android.webview.WebViewActivity
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class ManageShortcutsViewModel @Inject constructor(
+    private val integrationUseCase: IntegrationRepository,
+    application: Application
+) : AndroidViewModel(application) {
+
+    private val TAG = "ShortcutViewModel"
+    private lateinit var iconPack: IconPack
+    private var shortcutManager = application.applicationContext.getSystemService<ShortcutManager>()!!
+    val canPinShortcuts = Build.VERSION.SDK_INT >= Build.VERSION_CODES.O && shortcutManager.isRequestPinShortcutSupported
+    @RequiresApi(Build.VERSION_CODES.N_MR1)
+    var pinnedShortcuts = mutableStateOf(shortcutManager.pinnedShortcuts)
+        private set
+    @RequiresApi(Build.VERSION_CODES.N_MR1)
+    var dynamicShortcuts = mutableStateOf(shortcutManager.dynamicShortcuts)
+        private set
+
+    var entities = mutableStateMapOf<String, Entity<*>>()
+        private set
+    var selectedIcon1 = mutableStateOf(0)
+        private set
+    var shortcutLabel1 = mutableStateOf("")
+        private set
+    var shortcutDesc1 = mutableStateOf("")
+        private set
+    var shortcutPath1 = mutableStateOf("")
+        private set
+    var shortcutType1 = mutableStateOf("lovelace")
+        private set
+    var drawableIcon1 = mutableStateOf(AppCompatResources.getDrawable(application, R.drawable.ic_stat_ic_notification_blue))
+        private set
+    var deleteShortcut1 = mutableStateOf(false)
+        private set

```

Comment:
This should be abstracted to a Data class and use a mutable list of those data objects rather than manually iterative over them.

Reply:
@jpelgrom you were right about this and it was my own misunderstanding :) 

Thank you both for the help 🙏